source/FreeRTOS_DNS.c

```diff
@@ -452,12 +452,12 @@
  * @param [in] pxAddress address structure
  * @returns whether sending the data was successful
  */
-    static BaseType_t prvSendBuffer( const char * pcHostName,
-                                     TickType_t uxIdentifier,
-                                     Socket_t xDNSSocket,
-                                     struct freertos_sockaddr * pxAddress )
+    static uint32_t prvSendBuffer( const char * pcHostName,
+                                   TickType_t uxIdentifier,
+                                   Socket_t xDNSSocket,
+                                   struct freertos_sockaddr * pxAddress )
     {
-        BaseType_t xReturn = pdFAIL;
+        uint32_t xReturn = pdFAIL;
```

Comment:
When `xReturn` becomes unsigned, it normally changes to `uxReturn`.
Please also check other appearances of `uint32_t xReturn`.